File: 1483-rank-teams-by-votes/1483-rank-teams-by-votes.cpp
class Solution {
public:
    string rankTeams(vector<string>& votes) {
        string answer = votes[0]; 
        if(votes.size() == 1) return answer; 

        map<char,vector<int>> rankings; 

        for(char & c : votes[0]){
            rankings[c] = vector<int>(votes[0].length(),0); 
        }

        for(string& s : votes){
            for(int i = 0; i < s.length(); i++){
                rankings[s[i]][i]++; 
            }
        }

        sort(answer.begin(), answer.end(), [&](char& a, char& b){
            if(rankings[a] == rankings[b]) return a < b; 
            return rankings[a] > rankings[b]; 
        });

        //cout << answer; 


        return answer; 
    }
};

File: 1483-rank-teams-by-votes/1483-rank-teams-by-votes.java
class Solution {
    public String rankTeams(String[] votes) {
        Map<Character,int[]> hashMap = new HashMap<>(); 
        
        for(String s : votes){
            for(int i = 0; i < s.length(); i++){
                hashMap.computeIfAbsent(s.charAt(i), k -> new int[votes[0].length()])[i]++; 
            }
        }

        char[] toChar = votes[0].toCharArray(); 
        List<Character> lst = new ArrayList<>(); 
        for(char c : toChar) lst.add(c); 

        Collections.sort(lst, (a,b) -> {
            int[] rankA = hashMap.get(a); 
            int[] rankB = hashMap.get(b); 
            for(int i = 0; i < votes[0].length(); i++){
                if(rankA[i] != rankB[i]){
                    return rankB[i] - rankA[i]; 
                }
            }

            return a - b; 
        }); 

        StringBuilder sb = new StringBuilder(); 
        for(char c : lst){
            sb.append(c); 
        }

        return sb.toString(); 
    }
}